python/tchannel/scheme.py

```diff
@@ -41,10 +62,18 @@ def __init__(self):
     def type(self):
         return self._type
 
-    def deserialize(self, obj):
+    def deserialize_header(self, obj):
+        assert obj
```

Comment:
Although in this specific case, we should try to handle as many cases as we can. @junchaowu consider returning `None` if `obj` is `None` or an empty string.